### PR TITLE
add delete interview

### PR DESCRIPTION
**Pull request by DfernandezNipo**
.

### CODE REVIEW

File: Library/Services/Implementation/NfieldInterviewService.cs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        private static void CheckSurveyId(string surveyId)
+        {
+            Ensure.ArgumentNotNullOrEmptyString(surveyId, nameof(surveyId));
+            if (!Guid.TryParse(surveyId,out var _))
+                throw new ArgumentException("surveyId is not a valid identifier");
+        }
+
+        private INfieldHttpClient Client => ConnectionClient.Client;
+
+        private string DeleteInterviewsApiUri(string surveyId, int interviewId)
```
Comment (hcsauniere) [inline Library/Services/Implementation/NfieldInterviewService.cs]: Better just "InterviewsApiUri"?

File: Library/Services/Implementation/NfieldInterviewService.cs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+using Nfield.Models;
+using Nfield.Models.NipoSoftware.Nfield.Manager.Api.Models;
+using Nfield.Utilities;
+
+namespace Nfield.Services.Implementation
+{
+    /// <summary>
+    /// Implementation of <see cref="INfieldInterviewQualityService"/>
+    /// </summary>
+    internal class NfieldInterviewService : INfieldInterviewService, INfieldConnectionClientObject
```
Comment (hcsauniere) [inline Library/Services/Implementation/NfieldInterviewService.cs]: for consistency "NfieldInterviewsService" (plural)

File: Library/Services/INfieldInterviewService.cs [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+//    along with Nfield.SDK.  If not, see <http://www.gnu.org/licenses/>.
+
+using System.Threading.Tasks;
+
+namespace Nfield.Services
+{
+    /// <summary>
+    /// Service to delete interview data
+    /// </summary>
+    public interface INfieldInterviewService
```
Comment (hcsauniere) [inline Library/Services/INfieldInterviewService.cs]: for consistency "INfieldInterviewsService" (plural)

File: Tests/Services/NfieldInterviewTests.cs [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+using Newtonsoft.Json;
+using Nfield.Infrastructure;
+using Nfield.Models;
+using Nfield.Models.NipoSoftware.Nfield.Manager.Api.Models;
+using Nfield.Services.Implementation;
+using Xunit;
+
+namespace Nfield.Services
+{
+    public class NfieldInterviewServiceTests : NfieldServiceTestsBase
```
Comment (hcsauniere) [inline Tests/Services/NfieldInterviewTests.cs]: for consistency "NfieldInterviewServicesTests " (plural)

File: Library/Extensions/NfieldConnectionClientExtensions.cs [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+                .ContinueWith(response => response.Result.Content.ReadAsStringAsync())
+                .Unwrap()
+                .ContinueWith(content =>
+                {
+                    var obj = JObject.Parse(content.Result);
+                    var status = obj["Status"].Value<int>();
+
+                    switch (status)
+                    {
+                        case 0: // pending
```
[the remark this answers is not in the document]
Comment (DfernandezNipo) [reply Library/Extensions/NfieldConnectionClientExtensions.cs]: I haven't written the code, only moved it.

**Pull request merged**